config.go

```diff
@@ -218,15 +219,21 @@ func (c *Config) defaultRegions() error {
 		return nil
 	}
 
-	if s := os.Getenv("AWS_REGION"); s != "" {
-		log.Debugf("region from AWS_REGION %q", s)
-		c.Regions = append(c.Regions, s)
+	s, err := session.NewSessionWithOptions(session.Options{
+		SharedConfigState: session.SharedConfigEnable,
+	})
+	if err != nil {
+		return err
+	}
```

Comment:
awesome, I didn't know this was in the SDK! 

small nitpicks:

- newline before `if err != nil` and after the closing brace
- `return errors.Wrap(err, "creating session")` (or similar)